common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1077,13 +1077,21 @@ public UTF8String translate(Map<Character, Character> dict) {
    * @return true if the parsing was successful else false
    */
   public boolean toLong(LongWrapper toLongResult) {
+    int numBytes = this.numBytes;
+    int offset = 0;
+    while (offset < numBytes && getByte(offset) == 0x20) {
+      offset++;
+    }
+    while (numBytes - 1 > offset && getByte(numBytes - 1) == 0x20) {
+      numBytes--;
```

Comment:
this is not numBytes, but the endIndex. numBytes should be endIndex - startIndex.

Reply:
yes, you're right.